lib/connection/commands.js

```diff
@@ -484,8 +485,15 @@ Response.prototype.parse = function(options) {
     // Deserialize but keep the array of documents in non-parsed form
     var doc = this.bson.deserialize(document, _options);
 
+    if (doc instanceof Error) {
+      throw doc;
+    }
     // Get the documents
-    this.documents = doc.cursor[documentsReturnedIn];
+    if (doc.cursor) {
+      this.documents = doc.cursor[documentsReturnedIn];
+    } else {
+      throw new MongoError('Cursor not found');
```

Comment:
I see. This should handle the case of `doc.ok === 0`. But, perhaps you could still use `errmsg`? Maybe something like ` throw new MongoError(doc.errmsg || 'Cursor not found')`?

Reply:
I agree with this. We could do `throw new MongoError(doc.errmsg ? doc : 'Cursor not found');`, so we could catch any error codes on the doc.